File: web/src/main/java/pl/spring/demo/web/rest/BookRestService.java
package pl.spring.demo.web.rest;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

import pl.spring.demo.service.BookService;
import pl.spring.demo.to.BookTo;

@Controller
@RequestMapping("/books")
public class BookRestService {

    @Autowired
    private BookService bookService;

    @RequestMapping(value = "/books-by-title", method = RequestMethod.GET)
    public @ResponseBody List<BookTo> findBooksByTitle(@RequestParam("titlePrefix") String titlePrefix) {
        return bookService.findBooksByTitle(titlePrefix);
    }

    @RequestMapping(value = "/book", method = RequestMethod.POST)
    public @ResponseBody BookTo saveBook(@RequestBody BookTo book) {
        return bookService.saveBook(book);
    }
    
    @RequestMapping(value = "/book", method = RequestMethod.DELETE)
    public void removeBook(@RequestBody Long bookId) {
    	bookService.removeBook(bookId);
    }
    
    @RequestMapping(value = "/book", method = RequestMethod.PUT)
    public @ResponseBody BookTo updateBook(@RequestBody BookTo book) {
    	bookService.updateBook(book);
    	return book;
    }
}

File: web/src/main/client/app/books/book.rest.service.js
angular.module('app.books').factory('bookRestService', function ($http, currentContextPath) {
    'use strict';

    return {
        search: function (titlePrefix) {
            return $http.get(currentContextPath.get() + 'services/books/books-by-title', {params: {titlePrefix: titlePrefix}});
        },
        deleteBook: function (bookId) {
        	console.log("[Delete] BookId: " + bookId);
            return $http.delete(currentContextPath.get() + 'services/books/book', {params: {bookId: bookId}});
        },
        addBook: function(book) {
        	return $http.post(currentContextPath.get() + 'services/books/book', book);
        },
        updateBook: function(book) {
        	return $http.put(currentContextPath.get() + 'services/books/book', book);
        }
    };
});
